api/runnerpool/ch_placer.go

```diff
@@ -30,28 +31,34 @@ func NewCHPlacer() Placer {
 // the LB_WAIT to drive placement decisions: runners only accept work if they have the capacity for it.
 func (p *chPlacer) PlaceCall(rp RunnerPool, ctx context.Context, call RunnerCall) error {
 
+	startTime := time.Now()
 	log := common.Logger(ctx)
 
 	// The key is just the path in this case
 	key := call.Model().Path
 	sum64 := siphash.Hash(0, 0x4c617279426f6174, []byte(key))
 
+OutTries:
 	for {
 		runners, err := rp.Runners(call)
 		if err != nil {
+			stats.Record(ctx, errorPoolCountMeasure.M(0))
 			log.WithError(err).Error("Failed to find runners for call")
 		} else {
+			if len(runners) == 0 {
+				stats.Record(ctx, emptyPoolCountMeasure.M(0))
```

Comment:
Should this code `break OutTries`?  It would save you from incrementing a bunch of the subsequent metrics, which may not be applicable in the case no runners exist.  E.g. do we want to record full-scan for a zero length runner list?